pkg/gardenlet/controller/backupentry/reconciler.go

```diff
@@ -599,19 +621,34 @@ func (r *Reconciler) newExtensionComponent(log logr.Logger, backupEntry *gardenc
 	)
 }
 
-func (r *Reconciler) waitUntilBackupBucketReconciled(ctx context.Context, log logr.Logger, backupBucket *gardencorev1beta1.BackupBucket) error {
-	return extensions.WaitUntilObjectReadyWithHealthFunction(
-		ctx,
-		r.GardenClient,
-		log,
-		health.CheckBackupBucket,
-		backupBucket,
-		"BackupBucket",
-		DefaultInterval,
-		DefaultSevereThreshold,
-		DefaultTimeout,
-		nil,
-	)
+func (r *Reconciler) checkIfBackupBucketIsHealthy(ctx context.Context, backupBucket *gardencorev1beta1.BackupBucket) error {
```

Comment:
Why does this function skip so many checks that were performed earlier in [checkExtensionObject](https://github.com/gardener/gardener/blob/599e7c50b76cb9215864d88190122ace566f0f5c/pkg/utils/kubernetes/health/extensions.go#L65)?

Reply:
The timestamp check is not required because we set the timestamp for extensions and not for the backupbucket, https://github.com/gardener/gardener/blob/55123e8724bca6939bb5a367d87ac432c0568c61/pkg/gardenlet/controller/seed/seed/reconciler_reconcile.go#L1031-L1046
Also we were retrying until a timeout and wanted to make sure we're working on a stale object.

I added the check for the operation annotation and observed generation.